src/pushsource/_impl/model/ami.py

```diff
@@ -41,6 +41,17 @@ class AmiRelease(object):
     """Release type, typically "ga" or "beta"."""
 
 
+@attr.s()
+class AmiBillingCodes(object):
+    """Billing codes associated with an AMI."""
+
+    name = attr.ib(type=str, default=None, validator=instance_of_str)
+    """Billing codes name, for example: Hourly2, arbitrary string for making image name unique"""
+
+    codes = attr.ib(type=list, default=None, validator=optional(instance_of(list)))
```

Comment:
Can you please use frozenlist for this to ensure the object remains immutable?

There are examples in other classes, like "content_types" in ErratumPushItem.  Basically you want to set 'default' and 'converter'.